mailpoet/lib/Automation/Engine/Storage/WorkflowStatisticsStorage.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Engine\Storage;
+
+use MailPoet\Automation\Engine\Data\WorkflowRun;
+use MailPoet\Automation\Engine\Data\WorkflowStatistics;
+
+class WorkflowStatisticsStorage {
```

Comment:
@johnolek Just adding my thoughts, I guess you're right, and it's not a very big class, so I guess it's OK to have it in `WorkflowRunStorage`. On the other hand, as @websupporter did it, it results in smaller classes, which is nice.

I think in the Doctrine code, we also might have some more specific repositories acting on the same table as other repositories, I guess it's fine if the case makes sense. How would you guys like just renaming this class to `WorkflowRunStatisticsStorage` or `WorkflowRunStatsStorage`? Having the "run" there, it could make more sense.

Reply:
I would not _yet_ want to rename it to WorkflowRunStatistics. I made it a new class, as I imagine the entity will grow a bit, once we have more detailed statistics. I think in the end, the decision on the best abstraction etc. is probably made, once we have a broader insight in what the statistics all need to serve instead of just the totals and what is the best way to achieve that.

Currently, my thinking for naming is: These are the statistics for a Workflow. From a `WorkflowRunStatistic` I would expect statistics for a specific run like maybe `started_at`, `steps_done`... The Storage has been named accordingly. So for me the naming of the storage is not derived from the tables it queries but the entities it provides.